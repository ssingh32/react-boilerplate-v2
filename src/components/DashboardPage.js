import React from 'react';

export const DashboardPage = () => (
  <div>
    Dashboard content
  </div>
);

export default DashboardPage;
