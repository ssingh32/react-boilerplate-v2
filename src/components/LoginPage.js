import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1>Boilerplate App</h1>
                <p>Tagline for App.</p>
                <button className="button" onClick={startLogin}>Login With Google</button>
            </div>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);