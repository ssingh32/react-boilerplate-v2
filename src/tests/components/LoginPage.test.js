import React from 'react';
import {LoginPage} from '../../components/LoginPage';
import { shallow } from 'enzyme';


test('should render Login Page correctly', () => {   
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
    let startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
  })