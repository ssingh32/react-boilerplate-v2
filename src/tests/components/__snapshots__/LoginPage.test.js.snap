// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1>
      Boilerplate App
    </h1>
    <p>
      Tagline for App.
    </p>
    <button
      className="button"
    >
      Login With Google
    </button>
  </div>
</div>
`;
